onnxruntime/core/framework/session_state.h

```diff
@@ -115,6 +116,24 @@ class SessionState {
    */
   const std::unordered_map<int, OrtValue>& GetConstantInitializedTensors() const;
 
+  /**
+  Get some initialized tensors (weights).
+  @param interested_weights The names of the weights to retrieve.
+  @param allow_missing_weights Whether to allow names in interested_weights
+         with no corresponding weight.
+  @param[out] retrieved_weights The retrieved weights.
+  @return The status of the operation.
+  */
+  Status GetInitializedTensors(
```

Comment:
Some of the new APIs in this class are not useful in inferencing mode. In fact they open up the once constant session state object to mutation. Can these be enclosed in the training mode macro so that it's clear why they're being added?